File: csharp/hamming/Hamming.cs
public static class Hamming
{
    public static int Compute(string firstStrand, string secondStrand)
    {
        var distance = 0;

        for (var i = 0; i < firstStrand.Length; i++) 
        {
            if (firstStrand[i] != secondStrand[i])
                distance++;
        }

        return distance;
    }
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {
    fun compute(leftStrand: String, rightStrand: String): Int {
        require(leftStrand.length == rightStrand.length, {"leftStrand and rightStrand must be of equal length."})

        val pairs = leftStrand.zip(rightStrand)
        return pairs.count {it.first != it.second}
    }
}